File: websocket-server/pkg/Types/bus.go
package types

type BusData struct {
	Lat   float64 `json:"lat"`
	Long  float64 `json:"long"`
	Index int     `json:"index"`
}

type BusClose struct {
	Close bool `json:"close"`
}

type BusRequest struct {
	BusId    string   `json:"busId"`
	BusData  BusData  `json:"busData"`
	BusClose BusClose `json:"busClose"`
	Which    string   `json:"which"`
}

type BusMessage struct {
	Message string `json:"message"`
}

type BusUser struct {
	UserId string `json:"userId"`
}

type BusUserLocation struct {
	UserId string  `json:"userId"`
	Lat    float64 `json:"lat"`
	Long   float64 `json:"long"`
}

type BusResponse struct {
	BusId           string          `json:"busId"`
	BusData         BusData         `json:"busData"`
	BusMessage      BusMessage      `json:"busMessage"`
	BusUserList     []string        `json:"busUserList"`
	BusUserLocation BusUserLocation `json:"BusUserLocation"`
	Which           string          `json:"which"`
}

File: bus-trace-frontend/src/types/bus.ts
export interface Bus {
  ID: string
  name: string
  lat: number
  long: number
}

interface busClose {
  close: boolean
}

enum BusReqWhich {
  BusData = 'busData',
  BusClose = 'busClose'
}

export interface BusReq {
  busId: string
  busData: busData
  busClose: busClose
  which: BusReqWhich
}
interface busData {
  lat: number
  long: number
  index: number
}

interface busMessage {
  busId: string
  message: string
}

interface busUserLocation {
  userId: string
  lat: number
  long: number
}

enum BusResWhich {
  BusData = 'busData',
  BusMessage = 'busMessage',
  BusUserList = 'busUserList',
  BusUserLocation = 'busUserLocation'
}

export interface BusRes {
  busId: string
  busData: busData
  busMessage: busMessage
  busUserList: string[]
  busUserLocation: busUserLocation
  which: BusResWhich
}
